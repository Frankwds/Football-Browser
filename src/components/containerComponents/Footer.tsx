import React from "react";

/**
 * React class component displaying the footer of the application
 * 
 * @returns {React.Component}
 */
class Footer extends React.Component {
  render() {
    return (
      <div>
        <p>Â©Team 50, IT2810</p>
      </div>
    );
  }
}

export default Footer;
