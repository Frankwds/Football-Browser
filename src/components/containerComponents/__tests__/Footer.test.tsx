import ReactDOM from "react-dom";
import Footer from "../Footer";
import renderer from "react-test-renderer";
import { act } from "react-dom/test-utils";


let container: any;

beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  document.body.removeChild(container);
  container = null;
});

describe("Testing Footer", () => {
  it("should render without crashing", () => {
    act(() => {
      const div = document.createElement("div");
      ReactDOM.render(<Footer />, div);
    })
  });

  it("snapshot should be same as previous", () => {
    const tree = renderer
      .create(<Footer />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

