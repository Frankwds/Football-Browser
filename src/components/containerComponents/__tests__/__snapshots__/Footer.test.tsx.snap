// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Footer snapshot should be same as previous 1`] = `
<div>
  <p>
    Â©Team 50, IT2810
  </p>
</div>
`;
